sklearn/metrics/_scorer.py

```diff
@@ -451,6 +450,13 @@ def _passthrough_scorer(estimator, *args, **kwargs):
     return estimator.score(*args, **kwargs)
 
 
+@validate_params(
+    {
+        "estimator": [HasMethods("fit")],
+        "scoring": [str, callable, None],
```

Comment:
I think that we could replace `str` with a set of all supported scoring strings. We should call the keys of the `_SCORER` dictionary available in `sklearn/metrics/_scorer.py`.

Reply:
```suggestion
        "scoring": [set(_SCORERS.keys()), callable, None],
```

Reply:
I didn't realise that a set would be a valid constraint. Shouldn't we use the `StrOptions` constraint instead? Also, the constraints I proposed were based on the current parameters validation for the `metrics.get_scorer` function. Should I also update the constraints for the `metrics.get_scorer` function in another PR to be consistent?

Reply:
> Shouldn't we use the StrOptions constraint instead?

Yes, I edited the suggestion

> Should I also update the constraints for the metrics.get_scorer function in another PR to be consistent?

Reading again the validation of ``get_scorer``, I'd be in favor of using get_scorer_names.

Reply:
Alright, thanks! I'll open another PR for `get_scorer` then.

Reply:
I'm fine with moving the definitions of ``get_scorer`` and ```check_scoring`` after the definition of ``_SCORERS``. What do you think @glemaitre ?

Reply:
Yes, let's do that. It makes sense.